stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/utils/CustomObjectFiltering.kt

```diff
@@ -55,12 +55,26 @@ internal fun <T : CustomObject> FilterObject.filter(t: T): Boolean = try {
         is LessThanOrEqualsFilterObject ->
             compare(t.getMemberPropertyOrExtra(fieldName, value::class) as? Comparable<Any>, value as? Comparable<Any>) { it <= 0 }
         is InFilterObject -> when (fieldName) {
-            MEMBERS_FIELD_NAME -> t.getMembersId().fold(false) { isIn, memberId -> isIn || values.contains(memberId) }
-            else -> values.contains(t.getMemberPropertyOrExtra(fieldName, Any::class))
+            MEMBERS_FIELD_NAME -> values.any(t.getMembersId()::contains)
+            else -> {
+                val fieldValue = t.getMemberPropertyOrExtra(fieldName, Any::class)
+                if (fieldValue is List<*>) {
```

Comment:
What about other collections? You could use `fieldValue is Iterable<*>` instead of `fieldValue is List<*>`